gateway/ln-gateway/src/rpc/lnrpc_client.rs

```diff
@@ -40,3 +47,51 @@ impl DynLnRpcClient {
         DynLnRpcClient(client)
     }
 }
+
+/// An `ILnRpcClient` that wraps around `GatewayLightningClient` for convenience,
+/// and makes real RPC requests over the wire to a remote lightning node.
+/// The lightnign node is exposed via a corresponding `GatewayLightningServer`.
+#[derive(Debug)]
+pub struct NetworkLnRpcClient {
+    client: GatewayLightningClient<Channel>,
+}
+
+impl NetworkLnRpcClient {
+    pub async fn new(url: Url) -> Result<Self> {
+        let endpoint = Endpoint::from_shared(url.to_string()).map_err(|e| {
+            error!("Failed to create lnrpc endpoint from url : {:?}", e);
+            LnGatewayError::Other(anyhow!("Failed to create lnrpc endpoint from url"))
+        })?;
+
+        let client = GatewayLightningClient::connect(endpoint)
+            .await
+            .map_err(|e| {
+                error!("Failed to connect to lnrpc server: {:?}", e);
+                LnGatewayError::Other(anyhow!("Failed to connect to lnrpc server"))
+            })?;
+
+        Ok(Self { client })
+    }
+}
+
+#[async_trait]
+impl ILnRpcClient for NetworkLnRpcClient {
+    async fn get_pubkey(&self) -> Result<GetPubKeyResponse> {
+        unimplemented!()
+    }
+
+    async fn pay_invoice(&self, _invoice: PayInvoiceRequest) -> Result<PayInvoiceResponse> {
```

Comment:
Shortened forms are neat. But seems limiting on the get side because we cannot then have `get(A)`, `get(B)`, no ?

Reply:
Opting to go with `long_snake_api()` definitions for now. We can polish names

Reply:
> Shortened forms are neat. But seems limiting on the get side because we cannot then have `get(A)`, `get(B)`, no ?

hm :thinking: don't understand what you mean